ultralytics/engine/trainer.py

```diff
@@ -762,3 +768,24 @@ def build_optimizer(self, model, name="auto", lr=0.001, momentum=0.9, decay=1e-5
             f'{len(g[1])} weight(decay=0.0), {len(g[0])} weight(decay={decay}), {len(g[2])} bias(decay=0.0)'
         )
         return optimizer
+
+    def _setup_hub(self):
+        """Setup session for Hub Training."""
+        from ultralytics.utils import SETTINGS
+
+        if SETTINGS["hub"] is False or self.hub_session is not None:
```

Comment:
I tried changing this to use the original logic:

```py
if SETTINGS["hub"] or not self.hub_session:
```

this showed during training:

```
2024-05-24 08:17:19,212 - hub_sdk.helpers.logger - ERROR - Unknown error occurred.
2024-05-24 08:17:19,213 - hub_sdk.helpers.logger - ERROR - Received no response from the server while creating the model.
```